### ISSUE TITLE
history data are never saved in sqlite database 

### ISSUE THREAD

**Issue by okyame** [FreeOpcUa/opcua-asyncio#60]
Hello,

I try do test history feature on the asyncio-server unsuccessfully.

I use the `server-datavalue-history.py` example script that I modify like this:

```
import asyncio
import sys
sys.path.insert(0, "..")
import time
import math


from asyncua import ua, Server
from asyncua.server.history_sql import HistorySQLite


async def main():

    # setup our server
    server = Server()
    
    # Configure server to use sqlite as history database (default is a simple memory dict)
    server.iserver.history_manager.set_storage(HistorySQLite("my_datavalue_history.sql"))
    
    # initialize server 
    await server.init()

    server.set_endpoint("opc.tcp://0.0.0.0:4840/freeopcua/server/")

    # setup our own namespace, not really necessary but should as spec
    uri = "http://examples.freeopcua.github.io"
    idx = await server.register_namespace(uri)

    # get Objects node, this is where we should put our custom stuff
    objects = server.get_objects_node()

    # populating our address space
    myobj = await objects.add_object(idx, "MyObject")
    myvar = await myobj.add_variable(idx, "MyVariable", ua.Variant(0, ua.VariantType.Double))
    await myvar.set_writable()  # Set MyVariable to be writable by clients
    print(myvar)

    # starting!
    await server.start()

    # enable data change history for this particular node, must be called after start since it uses subscription
    await server.historize_node_data_change(myvar, period=None, count=100)

    try:
        count = 0
        while True:
            time.sleep(1)
            count += 0.1
            await myvar.set_value(math.sin(count))
            print(count)

    finally:
        # close connection, remove subscriptions, etc
        await server.stop()

if __name__ == "__main__":
    loop = asyncio.get_event_loop()
    loop.set_debug(True)
    loop.run_until_complete(main())
```

The data are never saved in sqlite3 database.

Has someone already had this problem?

**Comment by okyame** [comment]
I find that, in history.py, in class `SubHandle` (line 185)  ,`datachange_notification` function is never called.

**Comment by oroulet** [comment]
if you add a print there, it is never printed? it is supposed to be tests for that method..

**Comment by okyame** [comment]
No, it's never printed.

**Comment by okyame** [comment]
When I stop the server with 'KeyboardInterrupt' shortcut, it calls `datachange_notification` before it stopped. But nothing is saved in database.

```
Node(NumericNodeId(ns=2;i=2))
Endpoints other than open requested but private key and certificate are not set.
Executing <Task pending coro=<main() running at examples/server-datavalue-history-asyncio.py:39> wait_for=<_GatheringFutur
e pending cb=[<TaskWakeupMethWrapper object at 0x7f40efd24858>()] created at /usr/lib/python3.6/asyncio/tasks.py:549> cb=[
_run_until_complete_cb() at /usr/lib/python3.6/asyncio/base_events.py:185] created at /usr/lib/python3.6/asyncio/base_even
ts.py:452> took 0.736 seconds
0.1
0.2
0.30000000000000004
^C
Executing <Handle <TaskWakeupMethWrapper object at 0x7f40efd246a8>(<Future finis...events.py:284>) created at /usr/lib/python3.6/asyncio/futures.py:375> took 3.472 seconds
Enter in subhander class !
Enter in subhander class !
Enter in subhander class !
Enter in subhander class !
Enter in save_node_value class !
Enter in save_node_value class !
Enter in save_node_value class !
Enter in save_node_value class !
Traceback (most recent call last):
  File "examples/server-datavalue-history-asyncio.py", line 60, in <module>
    loop.run_until_complete(main())
  File "/usr/lib/python3.6/asyncio/base_events.py", line 460, in run_until_complete
    self.run_forever()
  File "/usr/lib/python3.6/asyncio/base_events.py", line 427, in run_forever
    self._run_once()
  File "/usr/lib/python3.6/asyncio/base_events.py", line 1432, in _run_once
    handle._run()
  File "/usr/lib/python3.6/asyncio/events.py", line 145, in _run
    self._callback(*self._args)
  File "examples/server-datavalue-history-asyncio.py", line 47, in main
    time.sleep(1)
KeyboardInterrupt
```

**Comment by okyame** [comment]
Finally, i found and fixed the issue: ""conn_commit()" is never called.

I modify this function in the `history_sql.py ` file like this:

```
    async def _execute_sql(self, sql: str, params: Iterable = None):
        await self._loop.run_in_executor(None, self._conn.execute, sql, params or ())
        await self._loop.run_in_executor(None, self._conn.commit)
```

**Comment by oroulet** [comment]
can you make a PR? that is easy

**Comment by okyame** [comment]
Here is PR

**Comment by cbergmiller** [comment]
Fixed by #65
